sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -370,6 +384,26 @@ def _add_singleton_step(self, label, full_label, tag, input_step):
           PropertyNames.OUTPUT_NAME: PropertyNames.OUT}])
     return step
 
+  def run_Impulse(self, transform_node):
+    standard_options = (
+        transform_node.outputs[None].pipeline.options.view_as(StandardOptions))
+    if standard_options.streaming:
+      step = self._add_step(
+          TransformNames.READ, transform_node.full_label, transform_node)
+      step.add_property(PropertyNames.FORMAT, 'pubsub')
```

Comment:
Yes this needs to be generalized, across batch and streaming as we add support for this primitive on the runner side. This is captured in the tasks related to SDF